osu.Server.Spectator/GracefulShutdownManager.cs

```diff
@@ -39,12 +44,36 @@ private void shutdownSafely()
         foreach (var store in dependentStores)
             store.StopAcceptingEntities();
 
-        TimeSpan timeWaited = new TimeSpan();
+        performOnAllRooms(async r =>
+        {
+            await r.StartCountdown(new ServerShuttingDownCountdown
+            {
+                TimeRemaining = TIME_BEFORE_FORCEFUL_SHUTDOWN
+            });
+        }).Wait();
 
+        TimeSpan timeWaited = new TimeSpan();
         TimeSpan timeBetweenChecks = TimeSpan.FromSeconds(10);
+        bool finalNotificationSent = false;
 
         while (timeWaited < TIME_BEFORE_FORCEFUL_SHUTDOWN)
         {
+            TimeSpan timeRemaining = TIME_BEFORE_FORCEFUL_SHUTDOWN - timeWaited;

```

Comment:
Was this implementation chose due to ease of implementation? I would have thought that one countdown would ve sent to all rooms when shutdown started, and one would be sent to any users joining a room on a server which is shutting down (at the point of joining the room).

If that sounds too hard to implement we can go with what you have to start with.

Reply:
What part? This is only done to send the other notification 5 minutes before shutdown. It is _also_ sent to all rooms at the start of the countdown and also when they join the room.

Should I remove the countdown in this loop? I'm unsure of it existing.

Reply:
I think just one notification type is fine (removing the boolean), as per discord discussion.

Ignore the rest of what I said, I missed that countdowns were sent as part of the room.